src/tokens/SegmentedControl.tsx

```diff
@@ -26,6 +26,11 @@ function SegmentedControl({ values, selectedIndex, onChange }: Props) {
           style={[
             styles.button,
             { backgroundColor: index === selectedIndex ? Colors.greenUI : Colors.gray1 },
+            // Round the left and right sides of the first and last buttons respectively
+            index === 0 && { borderBottomEndRadius: 0, borderTopEndRadius: 0 },
+            index === values.length - 1 && { borderBottomStartRadius: 0, borderTopStartRadius: 0 },
+            // Square corners, no border radius, for middle buttons
+            index !== 0 && index !== values.length - 1 && { borderRadius: 0 },
```

Comment:
This looks different from previously. Did you check with @nityarao that it's what we want?